### DISCUSSION TITLE
Solutions

### DISCUSSION THREAD

**Discussion by Grandyanc** [community/community#40185] [New to GitHub] [answered]
### Select Topic Area

General

### Body

Hi there,
Am building a project, I have this page where users can buy a token from the site. When buying the token you will have to select the token name from the drop down option that fetched from the database, after selecting a name the price of the selected option will display in the text field, user will input their qualities and the total will auto calculate and display in the input field. I have userbuy, usersell, Token, as my table names In the database, the table name 'Token' is where user can buy the tokens from, Id, NameofToken, Price. In the userbuy table I have Id, TokenName, Amount, Quantity, Totalbuy, Image, Interest. After the user clicks on buy button the system will calculate a 3 percent interest from the Totalbuy for the user and insert it into the Interest column in the table, the order will go through if user has enough fund in his account else display s error of 'not enough fund'. This is how I want the page backend to work.

Please if you understand my problem, I just need an example working code to follow and do mine. Please any solution?

**Comment by jfullstackdev** [comment, marked as answer]
you mean you need help, as in coding ? did you already start the project by yourself, because you are mentioning the names of the tables

maybe you can be very specific with the `if else` situation

**Reply by Grandyanc** [reply]
Yes, I have already started the working on it but since am an amateur in PHP it's difficult to get the codes working. So I want a helping hand with a complete functional code based on my description so that I can follow to do mine. I listed the details of the database table to give an idea of what I mean. Thank you

**Reply by Grandyanc** [reply]
Thank you for your response, now let's say this is my database with 3 tables.

token - id, Name, Price,. 

users - Id, AcctNo, Balance, Username, Fullname, Email, Password, Role.

usertoken - id, AcctNo, TokenName, Amount, Quantity, Totalbuy, Interest.

The table named 'token' is where products are stored and registered users can buy. On the html form I have a drop-down option, a text field labelled 'Price', 'Quantity' and 'Total' with the button 'BUY', the drop-down option is where the token's names are fetched from the database, when a user selects a particular token he wants to buy and the price will display in the Price text field then the user will input the quantity in the Quantity text field and it will auto calculate in the Total text field, now, after the user clicks on buy button the system will calculate a 3 percent interest as a discount from the Totalbuy for the user this value will be stored in the 'Interest' column on the usertoken table.

When user submits form, system will check if user has enough fund in his 'Balance' else displays error of 'not enough fund'

**Reply by jfullstackdev** [reply]
first move is to not have the database, have some simple code to do that, you can use temporarily data structures, once you are getting results, it's time for you to write the actual code

database is just to save your work when the computer shuts down, it's simply the persistent data

when I was just starting, I never used an actual database, but in actual coding I started using it but there is not much difference, it's just there to save the data, say, from users
